site/content/how-to/traffic-management/snippets-filters.md

```diff
@@ -0,0 +1,345 @@
+
+```shell
+for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee; done
+```
+
+You should see some successful responses from the coffee Pod, however you should see multiple responses like the following:
+
+```text
+Request ID: 890c17df930ef1ef573feed3c6e81290
+<html>
+<head><title>503 Service Temporarily Unavailable</title></head>
+<body>
+<center><h1>503 Service Temporarily Unavailable</h1></center>
+<hr><center>nginx</center>
+</body>
+</html>
+```
+
+This is the default error response given by NGINX when the rate limit burst is exceeded, meaning our `SnippetsFilter`
+correctly applied our rate limiting NGINX configuration changes.
+
+## Configure rate limiting to the tea application
+
+Now, lets configure a different set of rate limiting rules to our tea application by adding the following `SnippetsFilter`:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsFilter
+metadata:
+  name: tea-rate-limiting-sf
+spec:
+  snippets:
+    - context: http
+      value: limit_req_zone $binary_remote_addr zone=teazone:10m rate=1r/s;
+    - context: http.server.location
+      value: limit_req zone=teazone burst=3;
+EOF
+```
+
+This `SnippetFilter` is the same as the one applied to the coffee HTTPRoute, however it removes the `nodelay` setting
+on the `limit_req` directive. This forces a delay on the incoming requests to match the rate set in `limit_req_zone`.
+
+Verify that the `SnippetsFilter` is Accepted:
+
+```shell
+kubectl describe snippetsfilters.gateway.nginx.org tea-rate-limiting-sf
+```
+
+You should see the following status:
+
+```text
+Status:
+  Controllers:
+    Conditions:
+      Last Transition Time:  2024-10-21T22:20:24Z
+      Message:               SnippetsFilter is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         gateway.nginx.org/nginx-gateway-controller
+Events:                      <none>
+```
+
+Verify that the tea `HTTPRoute` which had an `InvalidFilter` condition earlier, no longer has that condition.
+
+```shell
+kubectl describe httproutes.gateway.networking.k8s.io tea
+```
+
+You should see the following conditions:
+
+```text
+Conditions:
+      Last Transition Time:  2024-10-23T00:33:08Z
+      Message:               The route is accepted
+      Observed Generation:   2
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+      Last Transition Time:  2024-10-23T00:33:08Z
+      Message:               All references are resolved
+      Observed Generation:   2
+      Reason:                ResolvedRefs
+      Status:                True
+      Type:                  ResolvedRefs
+```
+
+Next, test that the `SnippetsFilter` is configured and has successfully applied the rate limiting NGINX configuration changes.
```

Comment:
```suggestion
Test that `SnippetsFilter` is configured and has successfully applied the rate limiting NGINX configuration changes.
```

Reply:
I'll leave `the` in place as I think thats correct since its an object and not a name.